src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -327,14 +335,49 @@ public static bool IsLinuxConsumptionOnAtlas(this IEnvironment environment)
                    string.IsNullOrEmpty(environment.GetEnvironmentVariable(LegionServiceHost));
         }
 
-        private static bool IsLinuxConsumptionOnLegion(this IEnvironment environment)
+        private static bool IsConsumptionOnLegion(this IEnvironment environment)
         {
             return !environment.IsAppService() &&
                    (!string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerName)) ||
                    !string.IsNullOrEmpty(environment.GetEnvironmentVariable(WebsitePodName))) &&
                    !string.IsNullOrEmpty(environment.GetEnvironmentVariable(LegionServiceHost));
         }
 
+        /// <summary>
+        /// Returns a value indicating whether the app is V1 Linux Consumption running on Legion.
+        /// </summary>
+        /// <param name="environment">The environment to verify.</param>
+        /// <returns><see cref="true"/> if the app is V1 Linux Consumption running on Legion; otherwise, <see cref="false"/>.</returns>
+        public static bool IsLinuxConsumptionOnLegion(this IEnvironment environment)
+        {
+            return IsConsumptionOnLegion(environment) && environment.WebsiteSkuIsDynamic();
```

Comment:
@balag0 Same question here - here we're also relying on SKU to make the determination. Is this correct and the best check we can be performing?

Reply:
based on my comment here
https://github.com/Azure/azure-functions-host/pull/10932#discussion_r2001235008

WEBSITE_SKU will be available in LinuxConsumption images only after specialization. using WEBSITE_SKU_NAME is better. but if we move the environment variables to docker files like @vivekjilla suggested, it would make things simpler/consistent

Reply:
But you shared with me the placeholder env vars for Linux Consumption on Legion [here](https://dev.azure.com/msazure/One/_git/AAPT-Antares-Websites?path=/src/Hosting/Functions/FunctionsPodService/PodBuilder/FunctionsWorkerPodBuilderStrategy.cs&version=GBdev&line=60&lineEnd=61&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), so SKU is already present and can be relied on as I'm doing here?